### ISSUE TITLE
Find in files searches files in memory even if the file on disk is modified

### ISSUE THREAD

**Issue by Stevehans** [notepad-plus-plus/notepad-plus-plus#6187]
### Description of the Issue
If a file on disk has been modified, find in files will not search the modified file, but rather the file in memory.

I think that if the file is modified on disk then this should be searched instead.

### Steps to Reproduce the Issue
1.  Open file A set to xyz, save
2.  Open file B, set the fred, save
3.  With focus on file B, edit file A on disk and change contents to abc
4. Search for xyz, in the results, file A is reported although it no longer contains this
5. Search for abc, file A is not shown in the results

### Expected Behavior
If a file is modified on disk then the on disk contents should be searched, not the in memory contents.

**Comment by sasumner** [comment]
This is a design decision that has already been made.  Here's a basic reason why it is the way it is:

When you run a *Find in Files* search, Notepad++ populates the *Find result* window.  Double-clicking an entry on the *Find result* window will take you to the double-clicked match.  The match needs to "match" what is textually in an open tab, since another copy of the tab (say from the disk version of the file) WILL NOT be opened.

**Comment by Stevehans** [comment]
But what happens is that clicking on the link takes you to the file, you're then prompted to reload as it's changed and if you reload you then don't find your string.

Maybe when searching it should take the file status into account and on the bar by the filename could say (only in memory) or (only on disk) for the search results, that would satisfy all eventualities?
Then if the file status indicates that the file on disk is different, both are searched.

**Comment by sasumner** [comment]
>you're then prompted to reload

You probably _DO NOT_ want to say Yes to the reload prompt at this point, as your in-memory changes are likely important and you don't want to lose them.

**Comment by Stevehans** [comment]
Not in this case, the files have been amended by a code generator so I do want them reloaded.

Don't think there's a one solution fits all without searching both files, some people (e.g. someone who has made a change but forgotten to save it) would probably want the in memory, someone else the on disk.

I think if you were able to incorporate the above change then it would satisfy all use cases?

Basically if the modified on disk flag is not set then do exactly as you do now, if set then search the in memory file and show that with the tag "In memory" and then the on disk one and tag that as "on disk", so if strings were found in both it would produce two sets in the results.

file.txt (in memory)
xyz
file.txt (on disk)
xyz
